### PR TITLE
En 5640 code coverage data retriever package

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: dataRetriever/resolvers/trieNodeResolver_test.go [hunk 1]
```diff
@@ -61,6 +61,7 @@ func TestNewTrieNodeResolver_OkValsShouldWork(t *testing.T) {
 
 	assert.NotNil(t, tnRes)
```
Comment (iulianpascalau) [inline dataRetriever/resolvers/trieNodeResolver_test.go]: this is tested on line 64

File: dataRetriever/resolvers/trieNodeResolver_test.go [hunk 2]
```diff
@@ -61,6 +61,7 @@ func TestNewTrieNodeResolver_OkValsShouldWork(t *testing.T) {
...
 	assert.Nil(t, err)
+	assert.False(t, tnRes.IsInterfaceNil())
```
Comment (iulianpascalau) [inline dataRetriever/resolvers/trieNodeResolver_test.go]: assert.False(t, check.IfNil(tnRes))

File: dataRetriever/shardedData/shardedData_test.go [hunk 3]
```diff
@@ -41,6 +42,29 @@ func TestNewShardedData_GoodConfigShouldWork(t *testing.T) {
 	sd, err := shardedData.NewShardedData(cacheConfigBad)
 	assert.Nil(t, err)
 	assert.NotNil(t, sd)
+	assert.False(t, sd.IsInterfaceNil())
```
Comment (iulianpascalau) [inline dataRetriever/shardedData/shardedData_test.go]: check.IfNil and remove assert on line 44

File: core/converters.go [hunk 4]
```diff
@@ -115,7 +115,7 @@ func EpochStartIdentifier(epoch uint32) string {
 // IsUnknownEpochIdentifier return if the epoch identifier represents unknown epoch
 func IsUnknownEpochIdentifier(identifier []byte) (bool, error) {
 	splitString := strings.Split(string(identifier), "_")
-	if len(splitString) == 0 || len(splitString[0]) == 0 {
+	if len(splitString) <= 1 || len(splitString[1]) == 0 {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply core/converters.go]: done.

File: dataRetriever/factory/metachain/resolversContainerFactory_test.go [hunk 5]
```diff
@@ -265,11 +265,12 @@ func TestNewResolversContainerFactory_ShouldWork(t *testing.T) {
 		&mock.Uint64ByteSliceConverterMock{},
 		&mock.DataPackerStub{},
 		createTriesHolder(),
-		0,
+		1,
 	)
 
 	assert.NotNil(t, rcf)
```
[the remark this answers is not in the document]
Comment (miiu96) [reply dataRetriever/factory/metachain/resolversContainerFactory_test.go]: refactored.

File: dataRetriever/requestHandlers/requestHandler_test.go [hunk 6]
```diff
@@ -83,6 +84,7 @@ func TestNewMetaResolverRequestHandler(t *testing.T) {
 	)
 	assert.Nil(t, err)
 	assert.NotNil(t, rrh)
```
[the remark this answers is not in the document]
Comment (miiu96) [reply dataRetriever/requestHandlers/requestHandler_test.go]: refactored

File: dataRetriever/requestHandlers/requestHandler_test.go [hunk 7]
```diff
@@ -535,6 +537,13 @@ func TestResolverRequestHandler_RequestMetaHeaderShouldCallRequestOnResolver(t *
 func TestResolverRequestHandler_RequestShardHeaderByNonceAlreadyRequestedShouldNotRequest(t *testing.T) {
 	t.Parallel()
 
+	defer func() {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply dataRetriever/requestHandlers/requestHandler_test.go]: refactored.

File: dataRetriever/dataPool/currentBlockPool_test.go [hunk 8]
```diff
@@ -0,0 +1,35 @@
+package dataPool
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
```
Comment (bogdan-rosianu) [inline dataRetriever/dataPool/currentBlockPool_test.go]: goimports

File: dataRetriever/errors.go [hunk 9]
```diff
@@ -13,6 +13,9 @@ var ErrNoConnectedPeerToSendRequest = errors.New("connected peers list is empty.
 // ErrNilValue signals the value is nil
 var ErrNilValue = errors.New("nil value")
 
+// ErrCannotFindTxInBlockPool signals the value is nil
+var ErrCannotFindTxInBlockPool = errors.New("cannot find tx in current block pool")
```
Comment (bogdan-rosianu) [inline dataRetriever/errors.go]: might rename to `ErrTxNotFoundInBlockPool`

File: dataRetriever/resolvers/transactionResolver_test.go [hunk 10]
```diff
@@ -106,6 +106,7 @@ func TestNewTxResolver_OkValsShouldWork(t *testing.T) {
 
 	assert.Nil(t, err)
 	assert.NotNil(t, txRes)
```
[the remark this answers is not in the document]
Comment (miiu96) [reply dataRetriever/resolvers/transactionResolver_test.go]: refactore.

File: dataRetriever/resolvers/trieNodeResolver_test.go (hunk 1 above)
Comment (miiu96) [reply dataRetriever/resolvers/trieNodeResolver_test.go]: refactored.

File: dataRetriever/resolvers/trieNodeResolver_test.go (hunk 2 above)
Comment (miiu96) [reply dataRetriever/resolvers/trieNodeResolver_test.go]: refactored.

File: dataRetriever/shardedData/shardedData_test.go (hunk 3 above)
Comment (miiu96) [reply dataRetriever/shardedData/shardedData_test.go]: refactored.

File: dataRetriever/dataPool/currentBlockPool_test.go (hunk 8 above)
Comment (miiu96) [reply dataRetriever/dataPool/currentBlockPool_test.go]: done.

File: dataRetriever/errors.go (hunk 9 above)
Comment (miiu96) [reply dataRetriever/errors.go]: renamed.

File: dataRetriever/requestHandlers/requestHandler_test.go [hunk 11]
```diff
@@ -553,6 +562,13 @@ func TestResolverRequestHandler_RequestShardHeaderByNonceAlreadyRequestedShouldN
 func TestResolverRequestHandler_RequestShardHeaderByNonceBadRequest(t *testing.T) {
 	t.Parallel()
 
+	defer func() {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply dataRetriever/requestHandlers/requestHandler_test.go]: Not done

[the remark this answers is not in the document]
Comment (miiu96) [reply dataRetriever/requestHandlers/requestHandler_test.go]: refactored

**Pull request merged**